### PR TITLE
[clause 27] Index all member functions in clause 27

### PR DESCRIPTION
**Pull request by AlisdairM**
Some work remains to properly index all of the free (namespace scope) functions.

Will review other clauses later, especially those clauses with lots of additions for C++17.

### CODE REVIEW

File: source/iostreams.tex [hunk 1]
```diff
@@ -1173,8 +1177,8 @@
 The format control information for both input and output.
 \end{itemdescr}
 
-\indexlibrary{\idxcode{fmtflags}!\idxcode{ios_base}}%
 \indexlibrary{\idxcode{ios_base}!\idxcode{fmtflags}}%
+\indexlibrary{\idxcode{fmtflags}!\idxcode{ios_base}}%
```
Comment (tkoeppe) [inline source/iostreams.tex]: Is this entry correct? It looks like the function is called `flags`, no` fmtflags`?!

File: source/iostreams.tex [hunk 2]
```diff
@@ -4335,7 +4405,8 @@
 explicit basic_istream(basic_streambuf<charT, traits>* sb);
 \end{itemdecl}
 
-\indexlibrary{\idxcode{init}!\idxcode{basic_ios}}%
+\indexlibrary{\idxcode{init}!\idxcode{basic_istream}}%
+\indexlibrary{\idxcode{basic_istream}!\idxcode{init}}%
```
Comment (tkoeppe) [inline source/iostreams.tex]: What is this entry indexing? I don't quite see where this belongs.

Comment (AlisdairM) [inline source/iostreams.tex]: Reverted to the original, it is indexing a reference to base class initialization in the following text.

File: source/iostreams.tex (hunk 1 above)
Comment (AlisdairM) [inline source/iostreams.tex]: Fixed - thanks, thought I had caught all of these (there were not many).

File: source/iostreams.tex [hunk 3]
```diff
@@ -13249,6 +13479,7 @@
 These functions enable use of \tcode{recursive_directory_iterator}
 with range-based for statements.
 
+\indexlibrary{\idxcode{begin}!\idxcode{recursive_directory_iterator}}%
```
Comment (tkoeppe) [inline source/iostreams.tex]: Why is there only one version of this index entry? Is this not a member function?

File: source/iostreams.tex [hunk 4]
```diff
@@ -13258,6 +13489,7 @@
 \returns \tcode{iter}.
 \end{itemdescr}
 
+\indexlibrary{\idxcode{begin}!\idxcode{recursive_directory_iterator}}%
```
Comment (tkoeppe) [inline source/iostreams.tex]: Double-check the name?

File: source/iostreams.tex (hunk 3 above)
Comment (AlisdairM) [inline source/iostreams.tex]: I believe this is for the free-function overloads.

File: source/iostreams.tex (hunk 4 above)
Comment (AlisdairM) [inline source/iostreams.tex]: Oops - good catch

File: source/iostreams.tex [hunk 5]
```diff
@@ -6029,7 +6114,7 @@
 defines a class that is responsible for doing exception safe prefix and suffix
 operations.
 
-\indexlibrary{\idxcode{sentry}!constructor}%
```
Comment (tkoeppe) [inline source/iostreams.tex]: Why are you deleting this here, but you left both "sentry" and "basic_ostream::sentry" for the destructor?

Comment (AlisdairM) [inline source/iostreams.tex]: As far as I can tell, I am consistent for constructor and destructor - the difference is between basic_istream::sentry and basic_ostream::sentry.  In one case, I retain the old index entry and add the new one, in the other, I just rename the old to the new.

I agree they should be consistent, but not sure which way we think is the best to resolve.  I mildly lean towards retaining both 'sentry!constructor' and 'basic_Xstream::sentry!constructor' (and similarly for destructors), but I lean the other way for 'operator bool' - with is terribly inconsistent of me again.

Preferences?

Comment (tkoeppe) [inline source/iostreams.tex]: None at all. We can leave it as is :-)

**Pull request merged**